From d116f4a9a4bcd12fe85c4679fd7cb663f5f9872b Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@eng.windriver.com>
Date: Wed, 13 Sep 2023 23:48:12 -0700
Subject: [PATCH] Replace distutils.sysconfig.get_python_lib() with
 sysconfig.get_path('purelib') - python 3.10 support

This is part of the effort for 51776 to get python 3.10 support.
This PR replace distutils.sysconfig.get_python_lib() with sysconfig.get_path('purelib')
as distutils has been deprecated by python 3.10.

Note sysconfig has been available since python 3.2.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

Upstream-Status: Backport from tensorflow github
[https://github.com/tensorflow/tensorflow/pull/51865]

Signed-off-by: Hongxu Jia <hongxu.jia@eng.windriver.com>
---
 tensorflow/api_template.__init__.py    | 5 ++---
 tensorflow/api_template_v1.__init__.py | 5 ++---
 2 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/tensorflow/api_template.__init__.py b/tensorflow/api_template.__init__.py
index 57676e93a68..8c328f601d8 100644
--- a/tensorflow/api_template.__init__.py
+++ b/tensorflow/api_template.__init__.py
@@ -26,7 +26,7 @@ only a placeholder to enable test cases to run. The TensorFlow build replaces
 this file with a file generated from [`api_template.__init__.py`](https://www.github.com/tensorflow/tensorflow/blob/master/tensorflow/api_template.__init__.py)
 """
 
-import distutils as _distutils
+import sysconfig as _sysconfig
 import importlib
 import inspect as _inspect
 import logging as _logging
@@ -116,8 +116,7 @@ _site_packages_dirs += [p for p in _sys.path if "site-packages" in p]
 if "getsitepackages" in dir(_site):
   _site_packages_dirs += _site.getsitepackages()
 
-if "sysconfig" in dir(_distutils):
-  _site_packages_dirs += [_distutils.sysconfig.get_python_lib()]
+_site_packages_dirs += [_sysconfig.get_path('purelib')]
 
 _site_packages_dirs = list(set(_site_packages_dirs))
 
diff --git a/tensorflow/api_template_v1.__init__.py b/tensorflow/api_template_v1.__init__.py
index 9684275e572..bd3bc65c90a 100644
--- a/tensorflow/api_template_v1.__init__.py
+++ b/tensorflow/api_template_v1.__init__.py
@@ -14,7 +14,7 @@
 # ==============================================================================
 """Bring in all of the public TensorFlow interface into this module."""
 
-import distutils as _distutils
+import sysconfig as _sysconfig
 import importlib
 import inspect as _inspect
 import os as _os
@@ -162,8 +162,7 @@ _site_packages_dirs += [p for p in _sys.path if "site-packages" in p]
 if "getsitepackages" in dir(_site):
   _site_packages_dirs += _site.getsitepackages()
 
-if "sysconfig" in dir(_distutils):
-  _site_packages_dirs += [_distutils.sysconfig.get_python_lib()]
+_site_packages_dirs += [_sysconfig.get_path('purelib')]
 
 _site_packages_dirs = list(set(_site_packages_dirs))
 
-- 
2.40.0

